UPGRADE.md

```diff
@@ -1454,6 +1454,91 @@ export const PostEdit = () => (
 
 Use the `<Title>` component instead.
 
+## No More Props Injection In `<SimpleForm>` and `<TabbedForm>`
+
+The form components used to clone their children, inspect their props to handle labels and inject the `margin` and `variant` props.
+
+This is no longer the case and makes custom layout easier.
+
+### `<SimpleForm>` and `<TabbedForm>` No Longer Clone Their Children
+
+This makes custom layouts easier as you don't need to worry about passing props through your intermediate components anymore:
+
+```diff
+-const LineWrapper = ({ children, ...props }) => (
+-    <div style="display: flex">
+-        {Children.map(children, child => cloneElement(child, { ...props, ...child.props }))}
+-    </div>
+-)
+
+const PostCreate = () => (
+    <Create>
+        <SimpleForm>
+-            <LineWrapper>
++            <div style="display: flex">
+                <TextInput source="title" />
+                <TextInput source="title" margin="normal" variant="outlined" />
+-            </LineWrapper>
++            </div>
+        </SimpleForm>
+    </Create>
+)
+```
+
+### `<SimpleForm>` and `<TabbedForm>` No Longer Accept `margin` and `variant`
+
+Just like Material-UI, we don't provide a way to specify those props at the form level. Instead, you can either set those props on the inputs or leverage the Material-UI [component overrides through theme](https://mui.com/customization/theme-components/) if you need to change them globally:
+
+```diff
+const PostCreate = () => (
+    <Create>
+-        <SimpleForm margin="normal" variant="outlined">
++        <SimpleForm>
+-            <TextInput source="title" />
++            <TextInput source="title" margin="normal" variant="outlined" />
+        </SimpleForm>
+    </Create>
+)
+```
+
+You have several options when leveraging the [theme component overrides](https://mui.com/customization/theme-components/):
```

Comment:
In line 1490 you already mention this link.
Maybe this entire line could be omitted

Reply:
I don't mind having links where they are needed :man_shrugging: